src/engraving/dom/noteline.cpp

```diff
@@ -0,0 +1,250 @@
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <https://www.gnu.org/licenses/>.
+ */
+
+#include "noteline.h"
+#include "linkedobjects.h"
+#include "factory.h"
+#include "note.h"
+#include "textline.h"
+
+namespace mu::engraving {
+static const ElementStyle noteLineStyle {
+    { Sid::noteLinePlacement,                  Pid::PLACEMENT },
+    { Sid::noteLineFontFace,                   Pid::BEGIN_FONT_FACE },
+    { Sid::noteLineFontFace,                   Pid::CONTINUE_FONT_FACE },
+    { Sid::noteLineFontFace,                   Pid::END_FONT_FACE },
+    { Sid::noteLineFontSize,                   Pid::BEGIN_FONT_SIZE },
+    { Sid::noteLineFontSize,                   Pid::CONTINUE_FONT_SIZE },
+    { Sid::noteLineFontSize,                   Pid::END_FONT_SIZE },
+    { Sid::noteLineFontStyle,                  Pid::BEGIN_FONT_STYLE },
+    { Sid::noteLineFontStyle,                  Pid::CONTINUE_FONT_STYLE },
+    { Sid::noteLineFontStyle,                  Pid::END_FONT_STYLE },
+    { Sid::noteLineAlign,                      Pid::BEGIN_TEXT_ALIGN },
+    { Sid::noteLineAlign,                      Pid::CONTINUE_TEXT_ALIGN },
+    { Sid::noteLineAlign,                      Pid::END_TEXT_ALIGN },
+    { Sid::noteLineFontSpatiumDependent,       Pid::TEXT_SIZE_SPATIUM_DEPENDENT },
+};
+
+Sid NoteLineSegment::getPropertyStyle(Pid pid) const
+{
+    if (pid == Pid::OFFSET) {
+        return Sid::NOSTYLE;
+    }
+    return TextLineBaseSegment::getPropertyStyle(pid);
+}
+
+NoteLineSegment::NoteLineSegment(Spanner* sp, System* parent)
+    : TextLineBaseSegment(ElementType::NOTELINE_SEGMENT, sp, parent, ElementFlag::MOVABLE)
+{
+}
+
+EngravingItem* NoteLineSegment::propertyDelegate(Pid pid)
+{
+    if (pid == Pid::NOTELINE_PLACEMENT) {
+        return noteLine();
+    }
+
+    return TextLineBaseSegment::propertyDelegate(pid);
+}
+
+Sid NoteLine::getPropertyStyle(Pid pid) const
+{
+    if (pid == Pid::OFFSET) {
+        return Sid::NOSTYLE;
+    }
+    return TextLineBase::getPropertyStyle(pid);
+}
+
+NoteLine::NoteLine(EngravingItem* parent)
+    : TextLineBase(ElementType::NOTELINE, parent, ElementFlag::MOVABLE)
+{
+    setAnchor(Spanner::Anchor::NOTE);
+    setDiagonal(true);
+    setSystemFlag(false);
+    setPlacement(PlacementV::ABOVE);
+
+    initElementStyle(&noteLineStyle);
+
+    setBeginText(u"");
+    setContinueText(u"");
+    setEndText(u"");
+    setBeginTextOffset(PointF(0, 0));
+    setContinueTextOffset(PointF(0, 0));
+    setEndTextOffset(PointF(0, 0));
+    setLineVisible(true);
+
+    setBeginHookType(HookType::NONE);
+    setEndHookType(HookType::NONE);
+    setBeginHookHeight(Spatium(1.5));
+    setEndHookHeight(Spatium(1.5));
+    setGapBetweenTextAndLine(Spatium(0.5));
+
+    setLineEndPlacement(NoteLineEndPlacement::OFFSET_ENDS);
+
+    resetProperty(Pid::BEGIN_TEXT_PLACE);
+    resetProperty(Pid::CONTINUE_TEXT_PLACE);
+    resetProperty(Pid::END_TEXT_PLACE);
```

Comment:
If these properties don't have an associated style, they should at least use `PropertyDefault` here, or indeed do a `resetProperty` (cause now you're duplicating these constants here and in `PropertyDefault`). You could make an array with all these properties and reset them all in one loop